core/typeSugar.ml

```diff
@@ -2262,6 +2265,7 @@ let type_pattern ?(linear_vars=true) closed
     then Types.fresh_type_variable (lin_any, res_any)
     else Types.fresh_type_variable (lin_unl, res_any) in
 
+  (* NOTE: outdated comments? basic idea is still correct *)
```

Comment:
I suppose we should update the comment to reflect what the code is **really** doing...